bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt

```diff
@@ -164,12 +174,13 @@ class NativeBridge : StateObserver {
     }
 
     private fun deliverPendingReports() {
-        lock.lock()
         val filenameRegex = """.*\.crash$""".toRegex()
+        lock.lock()
```

Comment:
Can we do `withLock` here as well?

Reply:
We could, but it doesn't improve things much since we also have a `catch` in this block. So we would end up with a `withLock` block, and a `try {} catch {}` block as well.